Added  addExpense and addMonthlyIncome feature to new class

etschulte opened the pull request:
I made a new class and added addExpense and addMonthlyIncome methods to it to get started on the add monthly income and add expense features.

Tbusk committed:
Adding user account creation feature

User is populated with getters and setters for username and password.
EWalletApp CreateUser works for user creation.  To add users, the
username must not be taken and the password must be complex.

Tbusk committed:
Merge pull request #16 from Tbusk/account-creation

Adding user account creation feature

etschulte committed:
Made new class and added 2 methods to it

etschulte committed:
Added comments to methods

Tbusk committed:
Creating GUI for main application

GUI is setup using CardLayout with a menubar to nav between features with ease, similar to how it is done in JavaFX with scene switching.  Each of the feature panels needs to be built still.

Tbusk committed:
Adding pages for add item and import item

Panels have been created for adding items and importing items.  Estimates and reports UI still has to designed.

Tbusk committed:
Updating UI of home and add item page

The size and spacing of the UI is modified.

Tbusk committed:
Updating UI of estimate page

Adding basic UI to estimate page

etschulte committed:
Made ExpenserMain not abstract

Tbusk committed:
Merge pull request #20 from Tbusk/ExpenserMain

Added new class and two methods to it

etschulte committed:
Added Expense feature

catpho commented:
Are you sure you wanted to merge with the class before your forked repository? Just making sure since I made the mistake of pushing to wrong repository

Tbusk committed:
Updated UI of reports and implemented some functionality

Income reports, expense reports, and detailed reports is populated when items are added.  Item adding will work once User, Wage, and Expense are updated on main branch.  Completed requirements of those on my end for testing. Home page is updated when items are added as well.

Tbusk committed:
Changed month input for income from field to combobox

On the Add Income page, previously there was a JTextField accepting input for months with no restrictions. Now there is a JComboBox with dates in a specific format.

Tbusk committed:
Adding comments to file and making slight improvements

Adding comments to file to give insight into the workings of this file as well as making slight modifications to various things the IDE recommends fixing.

Tbusk committed:
Refactoring methods to make use of ExpenserMain

Methods that weren't using ExpenserMain that should are now using ExpenserMain.

Tbusk committed:
Adding methods useful for getting and setting information

Initializing the arraylists, adding expense variable, and adding various methods useful for updating variables and getting data used throughout the app.

Tbusk committed:
Adding Javadoc comments

Adding Javadoc comments to User.java.  Forgot to do it in last push.

etschulte committed:
Fixed the Expense feature

etschulte committed:
Updated methods to return values

etschulte committed:
Updated parameters

Tbusk committed:
Merge pull request #21 from Tbusk/ExpenserMain

Added Expense feature

Tbusk committed:
Updating ExpenserMain with methods

Filling out addExpense, addMonthlyIncome, and updateMonthlySavings

Tbusk committed:
Fixing merge conflict

Replaced ExpenserMain.java file with one from master.

Tbusk committed:
Reapplying User methods to ExpenserMain

addExpense, addMonthlyIncome, and updateMonthlySavings methods update

Tbusk committed:
Updating Spacing

Correcting spacing in file

Tbusk committed:
Refixing merge conflict

Merge is requiring file to be the exact same as in master

Tbusk committed:
Adding back data to ExpenserMain

Readding data to ExpenserMain and going to fix on GitHub

Tbusk committed:
Merge branch 'master' into report_gui

Tbusk committed:
Fixing method name conflicts in EWalletAPp

Names switched from test to repo versions.  Updated methods used from yearlyFrequency() to frequency().

Tbusk committed:
Fixing method name conflicts in EWalletApp and User

Minor tweaks

Tbusk committed:
Feature: Export to CSV + Income and Export Report Filtering

Big update: adding export to csv feature, filtering of income report and expense report, as well as income and expense related information for completing user stories.

Tbusk committed:
Merge pull request #22 from Tbusk/report_gui

GUI Version 1

etschulte committed:
Added methods to filter income and expense ArrayLists

Tbusk committed:
Merge pull request #23 from Tbusk/reports

Resolve #3 #4 #5 #6

Tbusk committed:
Tweaking GUI to use new methods

Temporary methods removed and replaced with new methods in ExpenserMain

Tbusk committed:
Merge pull request #24 from Tbusk/report_gui

Tweaking GUI to use new filter methods for tables

Tbusk committed:
Adding file import file reading

Adding file reading capabilty to software and wage object creation.

Tbusk committed:
Refactoring update table methods + updating table on import

Tweaking how tables are updated.  Creating new methods in ExpenserMain to handle updating of tables and values to reduce logic in EWalletApp.java.

Tbusk committed:
Updating type filter in income report panel upon import

The type filter in the income report panel will now populate when items are imported.

Tbusk committed:
Updating detailed reports + adding comments + dialog boxes for import page

Detailed reports page now contains imported income information.  There are now comments explaining whats going on.  Also, the buttons now contain basic logic with alerts if, for example, a user forgets to select a file, among others.

Tbusk committed:
Removed unused method

Unused method removed.

Tbusk committed:
Merge pull request #25 from Tbusk/import_income

New Feature: Import Income from csv file

Tbusk committed:
Adding feature: import expenses from csv + refactoring

Users can now import expenses from csv files.  Also, additional refactoring was done to improve readability and maintainability.  Comments were added.

Tbusk committed:
Merge pull request #26 from Tbusk/import_expense

Adding feature: import expenses from csv + refactoring

etschulte committed:
Added Login Page GUI

etschulte committed:
Fixed layout and added button

etschulte committed:
Added create account gui.

Tbusk committed:
Merge pull request #27 from Tbusk/login_gui

Added Login Page GUI

Tbusk committed:
Merge branch 'master' into createAccount_gui

Tbusk committed:
Merge pull request #29 from Tbusk/createAccount_gui

Create account gui closes #29

etschulte committed:
updated with new logins

etschulte committed:
Added functionality to create account gui and updated cvs file with test

etschulte committed:
tested create account feature

etschulte committed:
Added login feature

Tbusk committed:
Merge pull request #30 from Tbusk/createAccount_gui

Closes #30 Create account gui

Tbusk committed:
Added message dialogs + fixed login feature + cleared fields after creation and login

Tbusk committed:
Fixing account creation bug

Fixing a bug with account creation that resulted in fields not being read correctly.

Tbusk committed:
Merge pull request #31 from Tbusk/loginFeature

Login feature

Tbusk committed:
Removing comments

Tbusk committed:
Creating file containing the application requirements called requirements.md

Tbusk committed:
Adding project config files

Tbusk committed:
Renaming Project

Tbusk committed:
Restructuring project tree, switching from eclipse-based project to a maven project, adding maven required configuration and files

Tbusk committed:
changing back User Stories title after renaming issue in requirements.md

Tbusk committed:
Splitting up

Tbusk committed:
Switching from JPanel to JTabbedPane (original)

Tbusk committed:
Relocating to different packages

Tbusk committed:
encapsulating variables in Wage

Tbusk committed:
Renaming services used in LoginPanel for login action

Tbusk committed:
Fixing incorrect naming due to incorrect renaming, and moving to new package for ExpenserMain

Tbusk committed:
adding target folder to gitignore

Tbusk committed:
Creating months enum

Tbusk committed:
replacing months array with constant

Tbusk committed:
replacing months array with constant

Tbusk committed:
switching out ArrayList type for List returns for greater flexibility

Tbusk committed:
Fixing renaming issue in CreateAccountPanel

Tbusk committed:
creating new constant for expense frequency

Tbusk committed:
creating new service for filtering

Tbusk committed:
creating new service for report generation

Tbusk committed:
creating new service for totaling

Tbusk committed:
creating new interface to make it easier to use data in expense and wage classes.

Tbusk committed:
switching from hard-type to flexible list type, encapsulating variables, and making some lists final

Tbusk committed:
implementing interface, making variables final

Tbusk committed:
implementing interface, making variables final, applying camelCase to yearly frequency variable

Tbusk committed:
updating parameter names

Tbusk committed:
removing usage of user variable in expenserMain

Tbusk committed:
removing usage of user variable in expenserMain

Tbusk committed:
removing usage of user variable in expenserMain, removing storing of lists needlessly

Tbusk committed:
removing storing of lists, removing usage of expenserMain

Tbusk committed:
removing userAtHand variable. Replacing with one passed by constructor.  Removing exportReport, cleaning up updating methods

Tbusk committed:
misc cleanup in ImportPanel

Tbusk committed:
removing unnecessary lists and instantiation, misc cleanup

Tbusk committed:
making username and password temp rather than global

Tbusk committed:
consolidating getting expenses and wages in update methods

Tbusk committed:
fixing incorrect file path in UserCreationService

Tbusk committed:
updating method and variable names for better readability

Tbusk committed:
updating method calls

Tbusk committed:
removing excessive logic chains and coupling when not necessary for adding transactions

Tbusk committed:
moving button action logic to new methods

Tbusk committed:
fixing report file names

Tbusk committed:
moving button actions to new methods

Tbusk committed:
reorganizing imports

Tbusk committed:
moving button actions into new methods

Tbusk committed:
moving button action into new method

Tbusk committed:
moving button action into new method

Tbusk committed:
moving button actions into new methods

Tbusk committed:
moving button action into new method

Tbusk committed:
making income and expense reports show correct filtered details

Tbusk committed:
reduced menu action handling to one extensible method

Tbusk committed:
renaming monthlySavings to totalIncome to reflect usage better.

Tbusk committed:
fixing reporting user method calls and filtered balance display

Tbusk committed:
converting to a record

Tbusk committed:
adjusting to use records for Currency, Wage, and Expense

Tbusk committed:
moving files and adjusting to use records

Tbusk committed:
renaming and moving methods from ExpenserMain

Tbusk committed:
Removing unused interface Expenser

Tbusk committed:
adding method to update home panel totals

Tbusk committed:
passing home panel to import panel, renaming expenser

Tbusk committed:
passing home panel to calls, adding calls to update totals in various areas.

Tbusk committed:
moving classes

Tbusk committed:
adding method to update amount label

Tbusk committed:
adding class to handle importing data

Tbusk committed:
cleaning up importing action

Tbusk committed:
fixing file path issue with checking passwords from the csv.

Tbusk committed:
removing comments

Tbusk committed:
adding junit5 and assertj-swing as dependencies

Tbusk committed:
Adding names for gui items to run tests on

Tbusk committed:
Creating tests for adding income and expenses

Tbusk committed:
adding jackson dataformat csv dependency

Tbusk committed:
reformatting csv file

Tbusk committed:
cleaning username checker a bit

Tbusk committed:
cleaning user creation service a bit

Tbusk committed:
cleaning password checker a bit

Tbusk committed:
adjusting method call name in LoginPanel

Tbusk committed:
creating class to handle reading and writing to the credentials csv file

Tbusk committed:
creating record class to store credentials read from csv file

Tbusk committed:
removing unnecessary logic from create account button action

Tbusk committed:
refactor: march update 2025 (#32)
